### PR TITLE
feat: add edit profile button

### LINKED ISSUE #58: feat: add profile edit pop up ui
Use dialog -> https://tailwindcss-radix.vercel.app/#dialog

add fields for editing ->
- username
- email

### PR DESCRIPTION
**Pull request by ssarkar551**

### CODE REVIEW

**Comment by ssarkar551** [comment]
added edit profile button in Profile page
image
image

File: src/components/index.ts [hunk 1]
```diff
@@ -2,3 +2,4 @@ export { ProfileContainer } from "./Profile/ProfileContainer";
 export { LoginButton } from "./LoginButton";
 export { ThemeToggle } from "./ThemeToggle";
 export { Navbar } from "./Navbar";
+export { EditProfile } from "./EditProfile";
```
Comment (AyanavaKarmakar) [inline src/components/index.ts]: Move `EditProfile.tsx` inside Profile folder and remove it from here

```suggestion
```

File: package.json [hunk 2]
```diff
@@ -43,9 +46,11 @@
     "@trpc/react-query": "^10.7.0",
     "@trpc/server": "^10.7.0",
     "clsx": "^1.2.1",
+    "install": "^0.13.0",
     "next": "13.1.1",
     "next-auth": "^4.18.8",
     "next-themes": "^0.2.1",
+    "npm": "^9.3.0",
```
Comment (AyanavaKarmakar) [inline package.json]: Why?

File: package.json [hunk 3]
```diff
@@ -43,9 +46,11 @@
     "@trpc/react-query": "^10.7.0",
     "@trpc/server": "^10.7.0",
     "clsx": "^1.2.1",
+    "install": "^0.13.0",
```
Comment (AyanavaKarmakar) [inline package.json]: ???

File: package.json [hunk 4]
```diff
@@ -31,9 +31,12 @@
     "**/*": "prettier --write --ignore-unknown"
   },
   "dependencies": {
+    "@headlessui/react": "^1.7.7",
     "@next-auth/prisma-adapter": "^1.0.5",
     "@next/font": "^13.1.1",
     "@prisma/client": "^4.8.0",
+    "@radix-ui/react-dialog": "^1.0.2",
+    "@radix-ui/react-icons": "^1.1.1",
```
Comment (AyanavaKarmakar) [inline package.json]: Uninstall this package. copy the SVG from https://icons.radix-ui.com.Then, paste it in icons folder and export-import it from there

**Review by AyanavaKarmakar — CHANGES_REQUESTED** [review]

**Review by AyanavaKarmakar — COMMENTED** [review]
run `npm i` after uninstalling the packages

File: package.json (hunk 2 above)
Comment (ssarkar551) [reply package.json]: added by mistake. deleted

File: package.json (hunk 3 above)
Comment (ssarkar551) [reply package.json]: added by mistake. deleted

File: package.json (hunk 4 above)
Comment (ssarkar551) [reply package.json]: done

File: src/components/Profile/EditProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
+import { Transition } from "@headlessui/react";
+import * as DialogPrimitive from "@radix-ui/react-dialog";
```
Comment (AyanavaKarmakar) [inline src/components/Profile/EditProfile.tsx]:
```suggestion
import * as Dialog from "@radix-ui/react-dialog";
```

Comment (AyanavaKarmakar) [reply src/components/Profile/EditProfile.tsx]: CTRL + F Find and replace all

**Review by AyanavaKarmakar — CHANGES_REQUESTED** [review]

File: src/components/Profile/EditProfile.tsx (hunk 5 above)
Comment (ssarkar551) [reply src/components/Profile/EditProfile.tsx]: done

**Review by AyanavaKarmakar — APPROVED** [review]

**Review by Nishith-Savla — APPROVED** [review]

**Pull request merged**